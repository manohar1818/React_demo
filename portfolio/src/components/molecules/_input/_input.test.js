import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Input from './_input';
const mockOnChange = jest.fn();
const value="";
const label="input";
describe("Input component",() =>{
    test("Matches the snapshot",() => {
        const input = create(<Input onChange={mockOnChange} label={label} value={value}/>);
        expect(input.toJSON()).toMatchSnapshot();

    })
    test("input element is rendered",() => {
        const { getByTestId, getByText } = render(<Input onChange={mockOnChange} label={label} value={value}/>);
        expect(getByTestId("input")).toBeInTheDocument();
        
    });
   
   
   

});