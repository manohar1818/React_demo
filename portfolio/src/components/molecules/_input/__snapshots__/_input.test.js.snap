// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Matches the snapshot 1`] = `
<div
  data-testid="input"
>
  <h5
    className="MuiTypography-root MuiTypography-h5"
  >
    input
  </h5>
  <div
    className="makeStyles-styleInputFields-1"
    data-testid="inputField"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      data-testid="inputAtom"
      onClick={[Function]}
      required={true}
      style={
        Object {
          "color": "#000",
        }
      }
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        data-testid=""
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;
